Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp

```diff
@@ -405,18 +407,32 @@ static void webkitMediaCommonEncryptionDecryptProcessProtectionEvents(WebKitMedi
                 continue;
             }
 
-            initData = WebCore::InitData(mappedBuffer.data(), mappedBuffer.size());
-            GST_DEBUG_OBJECT(self, "init data of size %u", mappedBuffer.size());
+            const uint8_t* data = mappedBuffer.data();
+            size_t dataSize = mappedBuffer.size();
+            GRefPtr<GstBuffer> initDataBuffer = buffer;
+            Vector<uint8_t> decodedData;
+            if (!g_strcmp0(origin, "smooth-streaming") && WTF::base64Decode(reinterpret_cast<const char*>(data), dataSize, { decodedData })) {
+                data = decodedData.data();
+                dataSize = decodedData.size();
+                initDataBuffer = adoptGRef(gst_buffer_new_allocate(nullptr, dataSize, nullptr));
+                GstMappedBuffer mappedInitDataBuffer(initDataBuffer.get(), GST_MAP_WRITE);
+                if (!mappedInitDataBuffer)
+                    continue;
```

Comment:
I think we should replace this if with:
ASSERT(mappedInitBuffer);

A buffer we just created should be mappable unless catastrophe.